### ISSUE TITLE
Question about WaveUnpool

### ISSUE THREAD

**Issue by sonnguyen129** [Aaditya-Singh/SAFIN#9]
When implement WaveEncoder, I use pretrain vgg from torchvision:
```
class WaveEncoder(nn.Module):
    def __init__(self):
        super(WaveEncoder,self).__init__()
        self.pad = nn.ReflectionPad2d(1)
        self.relu = nn.ReLU(inplace=True)

        vgg = vgg19(pretrained=True).features
        self.conv1_1 = vgg[0]
        self.conv1_2 = vgg[2]
        self.pool1 = WavePool(64)
        self.conv2_1 = vgg[5]
        self.conv2_2 = vgg[7]
        self.pool2 = WavePool(128)
        self.conv3_1 = vgg[10]
        self.conv3_2 = vgg[12]
        self.conv3_3 = vgg[14]
        self.conv3_4 = vgg[16]
        self.pool3 = WavePool(256)
        self.conv4_1 = vgg[19]
```
The result is:
ll shape: torch.Size([25, 256, 39, 39])
lh,hl,hh shape: torch.Size([25, 256, 37, 37])
So catch a mismatch dimension error in WaveUnpool

In WaveUnpool, I see ```reshape``` function:
```
class WaveUnpool(nn.Module):
    def reshape(self, x, y):
        assert len(x.shape)==len(y.shape)==4
        y = y[:, :, :x.shape[2], :x.shape[3]]
        return y
```
Can you explain me the ```reshape``` function, while the third and fourth dimension of ll has bigger shape than lh,hl,hh. And how should I fix the bug. Thank you so much.

**Comment by Aaditya-Singh** [comment]
Kindly refer to [this issue](https://github.com/Aaditya-Singh/SAFIN/issues/3) for a detailed explanation of the `reshape` function. Note that we have also used a normalized VGG19 so it's possible that you don't need to import a pretrained one from torchvision, especially if you just want to use it for stylizing your images.
